src/Model/Contact.php

```diff
@@ -2844,6 +2861,8 @@ public static function removeFollower(array $contact)
 	 */
 	public static function removeSharer(array $contact)
 	{
+		self::clearFollowerFollowingEndpointCache($contact['uid']);
```

Comment:
Shouldn't this done after the contact was removed to avoid a possible race condition?